scripts/ci/backport_audit/utils.py

```
@@ -0,0 +1,289 @@
+    """
+    # Extract base version
+    match = re.match(r"release-(\d+\.\d+)", branch_name)
+    if not match:
+        msg = f"Invalid branch format: {branch_name}"
+        raise BackportAuditError(msg)
+
+    base_version = match.group(1)
+
+    current_version = None
+    try:
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)